cloudify_agent/installer/__init__.py

```diff
@@ -353,7 +367,24 @@ def download(self, url, destination=None):
                 url,
                 output_path=destination,
                 skip_verification=not verify_cert,
-                certificate_file=local_cert_file)
+                certificate_file=self._get_certificate_file())
 
     def move(self, source, target):
         self.runner.move(source, target)
+
+    def _get_certificate_file(self):
+        cert_file = self.cloudify_agent['local_rest_cert_file']
+        cert_content = self.cloudify_agent.get('rest_cert_content')
+
+        if not cert_content:
+            return None
+
+        if not self.runner.exists(cert_file):
+            with tempfile.NamedTemporaryFile(delete=False) as f:
+                f.write(cert_content.decode('string-escape'))
+            try:
+                self.runner.put_file(f.name, cert_file)
```

Comment:
The implicitness of copying a file during a simple `get` call somewhat bothers me. It will mostly surprise people that read this code. We can probably explicitly perform this step in `create_agent` and have `_get_certificate_file` only return a path to the file.